[CLD-232]: fix(addressbook): import CLDF directy for address book instead of type alias

graham-chainlink opened the pull request:
Remove the type alias to CLDF repository now that dependent repository like CLD are migrated off from using the type alias. Usage of addressbook in this repo now imports CLDF directly.

Main change is the deletion of [migrated_address_book.go](https://github.com/smartcontractkit/chainlink/blob/adde339a9dec3e29161dea5c00f859108448c3a5/deployment/migrated_address_book.go#L13-L13)

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-232


**NOTE: 
Lint failure is expected as i am touching a range of files and ci is flagging existing lint issues, using tag `allow-lint-issues` to bypass it**

graham-chainlink commented on deployment/.golangci.yml:
```diff
@@ -168,7 +168,7 @@ linters-settings:
   exclude-rules:
     # when performing simple refactor, the usage of addressbook will fail CI, this is a temporary ignore
-    - text: "SA1019: (.*).ExistingAddresses is deprecated: AddressBook is deprecated"
+    - text: "SA1019: (.*) is deprecated: AddressBook is deprecated"
```
This is needed since i modified a huge range of files and any existing lint errors will fail the ci

jmank88 replied:
We have a label for bypassing the linter in exceptional cases. Please use that instead of modifying the ruleset

graham-chainlink replied:
nice i did not know that, I added the `allow-lint-issues` , but seems like the [lint CI ](https://github.com/smartcontractkit/chainlink/actions/runs/14928316318/job/41938921329?pr=17640) still runs, did i do something wrong?

jmank88 replied:
It runs but the required job at the end won't fail

graham-chainlink replied:
Nice ! Got it working after a repush

graham-chainlink commented:
Lint failure is expected as i am touching a range of files and ci is flagging existing lint issues, using tag `allow-lint-issues` to bypass it

graham-chainlink committed:
fix(addressbook): remove alias

Remove type alias to CLDF repository now that dependent repository like CLD are migrated off from using the alias

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-232

jkongie commented on deployment/helpers.go:
```diff
@@ -88,3 +90,42 @@ func AddressListContainsEmptyAddress(addresses []common.Address) bool {
 	return false
 }
+
+func MigrateAddressBook(addrBook deployment.AddressBook) (datastore.MutableDataStore[datastore.DefaultMetadata,
```
Was this function moved or added back in for some reason?

jkongie replied:
Ignore this. I see it was moved from another file
